### PR TITLE
Implement bank grpc

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: x/bank/src/client/grpc.rs [hunk 1]
```diff
@@ -37,72 +40,118 @@ where
...
+            Err(Status::internal(ERROR_STATE_MSG))
+        }
     }
 
     async fn denom_owners(
         &self,
         _request: Request<QueryDenomOwnersRequest>,
     ) -> Result<Response<QueryDenomOwnersResponse>, Status> {
-        unimplemented!() //TODO: implement
+        Err(Status::internal(UNIMPLEMENTED_MSG))
```
Comment (NYBACHOK) [inline x/bank/src/client/grpc.rs:154]: I omited this route 'cause it implemented only in newer version of bank xmod.

File: gears/src/types/base/coins/unsigned.rs [hunk 2]
```diff
@@ -26,6 +28,68 @@ impl UnsignedCoins {
 
         true
     }
+
+    // TODO: Move this to generic declaration
+    /// Substracts matching coins. If the other coins have bigger values or the coins that don't
+    /// exists in original set, method returns error. If all coins are identical method returns
+    /// error.
+    pub fn checked_sub(&self, other: &UnsignedCoins) -> Result<Self, CoinsError> {
```
Comment (joneskm) [inline gears/src/types/base/coins/unsigned.rs:36]: Is there a test for this method?

File: gears/src/types/base/coins/unsigned.rs [hunk 3]
```diff
@@ -26,6 +28,68 @@ impl UnsignedCoins {
...
+        } else {
+            Err(CoinsError::InvalidAmount)
+        }
+    }
+
+    fn checked_calculate_iterate(
+        &self,
+        other_coins: &[UnsignedCoin],
+        operation: impl Fn(Uint256, Uint256) -> Result<Uint256, OverflowError>,
+    ) -> Result<Vec<UnsignedCoin>, CoinsError> {
```
Comment (joneskm) [inline gears/src/types/base/coins/unsigned.rs:54]: Same as above?

**Review by joneskm — APPROVED** [review]

File: gears/src/types/base/coins/unsigned.rs (hunk 2 above)
Comment (NYBACHOK) [reply gears/src/types/base/coins/unsigned.rs:36]: This method is a copy from  decimal implementation done by @dmvict , and it has tests. I made a quick copy for now of his code.

**Pull request merged**